templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -687,6 +686,33 @@ CRM.$(function($) {
 {/literal}{/if}{literal}
 
 CRM.$(function($) {
+  onPaymentMethodChange();
+  $("#payment_instrument_id").on("change",function(){
+    onPaymentMethodChange();
+  });
+
+  function onPaymentMethodChange() {
+    var paymentInstrument = $('#payment_instrument_id').val();
```

Comment:
@pradpnayak please do clarify the use case(s) here for us. @eileenmcnaughton my sense is that some organizations use a Point of Sale terminal to process the payment then they enter it manually in CiviCRM as a credit card payment. That is a case that has always been supported. There are also backoffice and frontoffice credit card submissions that should be determining the value of the type using the existing code that highlights the card type on the form after the card number has been entered. I wanted us to use that existing js functionality to provide the card type that would be stored.